instrumentation/spring/spring-data/spring-data-common/testing/src/main/java/AbstractSpringJpaTest.java

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.satisfies;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.sdk.testing.assertj.TraceAssert;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.List;
+import org.hibernate.Version;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public abstract class AbstractSpringJpaTest<
+    ENTITY, REPOSITORY extends JpaRepository<ENTITY, Long>> {
+
+  @RegisterExtension
+  private static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  abstract ENTITY newCustomer(String firstName, String lastName);
+
+  abstract Long id(ENTITY customer);
+
+  abstract void setFirstName(ENTITY customer, String firstName);
+
+  abstract Class<REPOSITORY> repositoryClass();
+
+  abstract REPOSITORY repository();
+
+  abstract List<ENTITY> findByLastName(REPOSITORY repository, String lastName);
+
+  abstract List<ENTITY> findSpecialCustomers(REPOSITORY repository);
+
+  void clearData() {
+    testing.clearData();
+  }
+
+  @Test
+  void testObjectMethod() {
+    REPOSITORY repo = repository();
+
+    testing.runWithSpan("toString test", repo::toString);
+
+    // Asserting that a span is NOT created for toString
+    testing.waitAndAssertTraces(
+        trace ->
+            trace
+                .hasSize(1)
+                .hasSpansSatisfyingExactly(
+                    span -> span.hasName("toString test").hasTotalAttributeCount(0)));
+  }
+
+  static void assertHibernate4Trace(TraceAssert trace, String repoClassName) {
+    trace
+        .hasSize(2)
+        .hasSpansSatisfyingExactly(
+            span ->
+                span.hasName("JpaCustomerRepository.save")
+                    .hasKind(SpanKind.INTERNAL)
+                    .hasAttributesSatisfyingExactly(
+                        equalTo(SemanticAttributes.CODE_NAMESPACE, repoClassName),
+                        equalTo(SemanticAttributes.CODE_FUNCTION, "save")),
+            span ->
+                span.hasName("INSERT test.JpaCustomer")
+                    .hasKind(SpanKind.CLIENT)
+                    .hasParentSpanId(trace.getSpan(0).getSpanId())
```

Comment:
I think you could use `hasParent(trace.getSpan(0))` instead

Reply:
Done, thanks for the feedback!